(QA-2501) add ability to install hosts in parallel

samwoods1 opened the pull request:
Run various parts of do_install in parallel based on the :install_in_parallel option impacting install_pe_on and upgrade_pe_on

puppetlabs-jenkins commented:
Can one of the admins verify this patch?

puppetlabs-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins-beaker.delivery.puppetlabs.net//job/qe_beaker-pe_btc-intn/7/

puppetlabs-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins-beaker.delivery.puppetlabs.net//job/qe_beaker-pe_btc-intn/15/

puppetlabs-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins-beaker.delivery.puppetlabs.net//job/qe_beaker-pe_btc-intn/22/

puppetlabs-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins-beaker.delivery.puppetlabs.net//job/qe_beaker-pe_btc-intn/34/

tvpartytonight commented on beaker-pe.gemspec:
```diff
@@ -32,6 +32,7 @@ Gem::Specification.new do |s|
 
   # Run time dependencies
   s.add_runtime_dependency 'stringify-hash', '~> 0.0.0'
+  s.add_runtime_dependency 'in-parallel', '~> 0.1'
```
Now that beaker itself has a runtime dependency requires `in-parallel`, does it make sense for beaker-pe to also specify the version? I think it would be cleaner to manage the the actual version in beaker itself, and then leave the dependency here with an unspecified version number, but with a comment about where that version is managed. What do you think?

jpartlow commented:
Has this been tested in a split configuration?  Because it seems like puppet agent does not get run on the master (because master is deleted from install_hosts), after the db and console node have been installed, which would be necessary to complete tk-auth configuration on the master.

puppetlabs-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins-beaker.delivery.puppetlabs.net//job/qe_beaker-pe_btc-intn/68/

puppetlabs-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins-beaker.delivery.puppetlabs.net//job/qe_beaker-pe_btc-intn/69/

puppetlabs-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins-beaker.delivery.puppetlabs.net//job/qe_beaker-pe_btc-intn/70/

samwoods1 commented:
@jpartlow hmm, I did test it and the install completed successfully, but you're right.  I intended for it to run puppet here https://github.com/puppetlabs/beaker-pe/pull/8/files#diff-921739f7679af755955835abf7807b64R438
I'll play with it and see why it seemed to work for me and update, or explain.

samwoods1 commented:
@jpartlow Good catch, I updated the code.  The reason it worked when I tested it is because I also had the agents doing frictionless installs.  This now works for split installs whether agents are frictionless or not.

puppetlabs-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins-beaker.delivery.puppetlabs.net//job/qe_beaker-pe_btc-intn/71/

puppetlabs-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins-beaker.delivery.puppetlabs.net//job/qe_beaker-pe_btc-intn/75/

samwoods1 commented:
I ran through the following tests locally:
MONO
 - master
 - agent
 - agent
MONO frictionless
 - master
 - agent, frictionless
 - agent, frictionless
SPLIT
 - master
 - database
 - dashboard
 - agent
 - agent
SPLIT frictionless
 - master
 - database
 - dashboard
 - agent, frictionless
 - agent, frictionless
MASTERLESS - made sure it errored.

puppetlabs-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins-beaker.delivery.puppetlabs.net//job/qe_beaker-pe_btc-intn/84/

samwoods1 commented:
I've run into a few things I'm working through... Trying to get this right.  I may need to talk with some people with more knowledge about the history of this stuff, but there are a few steps that do not appear to be doing anything... like if I remove them, everything still installs fine, and it speeds up the process, but I don't want to remove anything that is necessary for maybe older installs.

puppetlabs-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://jenkins-beaker.delivery.puppetlabs.net//job/qe_beaker-pe_btc-intn/111/

puppetlabs-jenkins commented:
Can one of the admins verify this patch?

kevpl commented:
@samwoods1 is it time to revitalize these changes?

puppetlabs-jenkins commented:
Can one of the admins verify this patch?

kevpl commented:
@samwoods1 this PR hasn't had activity in quite a while so I'm going to close it. Feel free to update the changes & submit a new PR about them
